Diagnostics/start-windows-build.ps1

```diff
@@ -0,0 +1,405 @@
+            'FilePath' = $programFile
+            'ArgumentList' = $prerequisites[$program]['install_args']
+            'Wait' = $true
+            'PassThru' = $true
+        }
+        if($programFile.EndsWith('.msi')) {
+            $parameters['FilePath'] = 'msiexec.exe'
+            $parameters['ArgumentList'] += @("/i", $programFile)
+        }
+        Write-Output "Installing $programFile"
+        $p = Start-Process @parameters
+        if($p.ExitCode -ne 0) {
+            Throw "Failed to install prerequisite $programFile during the environment setup"
+        }
+    }
+    # Add all the tools to PATH
+    $toolsDirs = @("$GOLANG_DIR\bin", "$GIT_DIR\cmd", "$GIT_DIR\bin", "$7ZIP_DIR")
+    $env:PATH += ';' + ($toolsDirs -join ';')
+    $env:GOPATH = $DIAGNOSTICS_DIR
+}
+
+function Start-DiagnosticsCIProcess {
+    Param(
+        [Parameter(Mandatory=$true)]
+        [string]$ProcessPath,
+        [Parameter(Mandatory=$false)]
+        [string[]]$ArgumentList,
+        [Parameter(Mandatory=$true)]
+        [string]$StdoutFileName,
+        [Parameter(Mandatory=$true)]
+        [string]$StderrFileName,
+        [Parameter(Mandatory=$true)]
+        [string]$BuildErrorMessage
+    )
+    $stdoutFile = Join-Path $DIAGNOSTICS_BUILD_LOGS_DIR $StdoutFileName
+    $stderrFile = Join-Path $DIAGNOSTICS_BUILD_LOGS_DIR $StderrFileName
+    New-Item -ItemType File -Path $stdoutFile -Force
+    New-Item -ItemType File -Path $stderrFile -Force
+    $logsUrl = Get-BuildLogsUrl
+    $stdoutUrl = "${logsUrl}/${StdoutFileName}"
+    $stderrUrl = "${logsUrl}/${StderrFileName}"
+    $command = $ProcessPath -replace '\\', '\\'
+    if($ArgumentList.Count) {
+        $ArgumentList | Foreach-Object { $command += " $($_ -replace '\\', '\\')" }
+    }
+    try {
+        Wait-ProcessToFinish -ProcessPath $ProcessPath -ArgumentList $ArgumentList `
+                             -StandardOutput $stdoutFile -StandardError $stderrFile
+        $msg = "Successfully executed: $command"
+    } catch {
+        $msg = "Failed command: $command"
+        $global:PARAMETERS["BUILD_STATUS"] = 'FAIL'
+        $global:PARAMETERS["LOGS_URLS"] += $($stdoutUrl, $stderrUrl)
+        $global:PARAMETERS["FAILED_COMMAND"] = $command
+        Write-Output "Exception: $($_.ToString())"
+        Throw $BuildErrorMessage
+    } finally {
+        Write-Output $msg
+        Write-Output "Stdout log available at: $stdoutUrl"
+        Write-Output "Stderr log available at: $stderrUrl"
+    }
+}
+
+function Get-LatestCommitID {
+    if(!$global:LATEST_COMMIT_ID) {
+        Throw "Failed to get the latest dcos-diagnostics commit ID. Perhaps it has not saved."
+    }
+    return $global:LATEST_COMMIT_ID
+}
+
+function Set-LatestDiagnosticsCommit {
+    Push-Location $DIAGNOSTICS_GIT_REPO_DIR
+    try {
+        if($CommitID) {
+            Start-ExternalCommand { git.exe reset --hard $CommitID } -ErrorMessage "Failed to set Mesos git repo last commit to: $CommitID."
```

Comment:
Ajust `ErrorMessage`. It says `Failed to set Mesos git repo` instead of `Failed to set Diagnostics git repo`